lib/clientsidescripts.js

```diff
@@ -47,24 +47,46 @@ function wrapWithHelpers(fun) {
  *     be passed as a parameter.
  */
 functions.waitForAngular = function(rootSelector, callback) {
```

Comment:
Do you think the logic would be easier to understand if we split this up?

```
var version = testForAngular()
...
if (version == 1) {
  waitForAngular1(...)
} else {
  waitForAngular2(...)
}
```

If someone less knowledgeable of angular2 (i.e. contributor) reads this, he/she would need to understand that angular1 contains `window.angular`, and angular2 contains `window.getAllAngularTestabilities()` and `window.getAngularTestability()`. This code is really hard to understand unless you know that.